### ISSUE TITLE
[feature request] Дадайце падтрымку .ро файлаў

### ISSUE THREAD

**Issue by k1llo** [BerlinerBY/Vily#2]
Прывітанне! Дзякуй за праграму. Адзін з найпапулярнейшых фарматаў файлаў перакладу гэта .ро файлы. Было б файна мець іх адразу,а не перапісваць парсер уручную. Насколькі я ведаю ў іх адзіны фармат.

**Comment by BerlinerBY** [comment]
Вітаю)
Першы раз чую пра такі. Пайду пачытаю. А вы можаце яшчэ і даслаць прыклады выкарыстання, каб адразу на іх патэсціў?

**Comment by k1llo** [comment]
Гэты фармат выкарыстоўваецца ў GTK праграмах пад Лінуксам у асноўным, вось прыклад: 
https://github.com/k1llo/forgetpass/tree/main/po
https://github.com/kra-mo/cartridges/tree/main/po
https://gitlab.com/mission-center-devs/mission-center/-/tree/main/po?ref_type=heads
Таксама ёсць біблятэка для працы з гэтымі файламі, напрыклад: https://pypi.org/project/potranslator/


Ну і так, я разумею, што гэта не зусім пра гульні, але ў перакладах звычайных праграм такога інструмента вельмі не хапае :)

**Comment by BerlinerBY** [comment]
Ага. Зразумеў як яно працуе. Вазьму ў распрацоўку, бо як раз яно патрабуе запланаваных змен у мадэлях. Падкажыце, калі ласка, а ці заўсёды ўся лакалізацыя змяшчаецца ў адным файле?

**Comment by k1llo** [comment]
Я ніколі не бачыў па-іншаму, таму ў 99.99% выпадкаў так :)
Тут ідэя ў тым што звычайна ё .po**T** ён як бы шаблон перакладу і з яго генеруецца .ро файл ужо з перакладам. Як гэта працуе я дакладна падказаць не магу. Хутчэй за ўсё на падставе гэтага файла проста ствараецца файл пераклада са змененымі метаданымі(іх можна знайсці ў пачатку кожнага файла) і адпаведнай назвай (be.po, en.po і г.д.)

**Comment by BerlinerBY** [comment]
Дадаў. Правярайце

**Comment by k1llo** [comment]
Выглядае так што працуе, не без памылак, але ўсё адно лепш чым нічога, аднак у мяне не адразу атрымалася запусціць яе, здаецца яна шукала нейкую часовую папку, але не знайшла і крашнулася, калі рукамі стварыў яе ўсё запрацавала.
З кропкі позірку UX у мяне ёсць некалькі пажаданняў:
1) Зрабіць запуск усяго адной камандай, бо трымаць 3 акна тэрмінала ну гэта не зручна
2) У ідэале зрабіць Docker вобраз з праграмай каб яе можна было задэплоіць
3) input_file запампоўваць праз браўзер і выбар xml альбо .ро рабіць таксама там

Я працягну тэсты і калі з'явяцца іншыя думкі ці ідэі як што рэалізаваць -- абавязкова напішу :)
Дзякуй за вашую працу!

**Comment by BerlinerBY** [comment]
Так, праблема з часовымі папкамі з'явілася з-за таго, што яны раней у .gitignore былі. Наконт пажаданняў, тут сапраўды варыянт з докерам зробіць жыццё крыху прасцей, а што да запампоўку, дык трэба паглядзець

**Comment by BerlinerBY** [comment]
Дарэчы, цікавы фармат файлаў. Але карыстаюцца ім як заўгодна. У вашых прыкладах msgid выкарыстоўваецца ў якасці id:
```
msgid "Site" 
msgstr ""
```
А зараз узяўся за другі праект і там у якасці id выкарыстоўваецца msgctxt, a msgid у якасці арыгінальнага радка:
```
msgctxt "achievement_1_1_title"
msgid "Don't Make Me Turn This Coach Around"
msgstr ""
```
З мінусаў фармату магу адзначыць, што для нашага тэкставага рэдактару не вельмі добра, калі ўся лакалізацыя знаходзіцца ў адным файле. Бо напрыклад зараз маю файл, які змяшчае 16 тысяч элементаў і ён пачынае падвісаць. Трэба ці то нейкую аптымізацыю пісаць, ці файл драбіць хаця б на кавалкі па ~5-6 тысяч элементаў.
Дарэчы, дадаў падтрымку .pot файлаў, але пакуль не каміціў
